### PR TITLE
Refactor resourcemanager tests to work with latest TestProxy

### PR DESCRIPTION
**Pull request by v-hongli1**
relate to [issue#40209](https://github.com/Azure/azure-sdk-for-java/issues/40209)

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java [hunk 1]
```diff
@@ -4,32 +4,40 @@
...
+import java.util.Set;
+
 public class VirtualMachineScaleSetEMSILMSIOperationsTests extends ComputeManagementTest {
     // LiveOnly because test needs to be refactored for storing/evaluating PrincipalId
     private String rgName = "";
-    private Region region = Region.US_WEST_CENTRAL;
+    private Region region = Region.US_WEST2;
     private final String vmssName = "javavmss";
+    private ResourceGroup resourceGroup;
+    VirtualMachineScaleSetSkuTypes virtualMachineScaleSetSkuTypes = VirtualMachineScaleSetSkuTypes.fromSkuNameAndTier("Standard_D2s_v3", "Standard");
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java:33]: private static final

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java [hunk 2]
```diff
@@ -976,9 +973,9 @@ public void canEnableMSIOnVirtualMachineScaleSetWithoutRoleAssignment() throws E
                 .computeManager
                 .virtualMachineScaleSets()
                 .define(vmssName)
-                .withRegion(region)
+                .withRegion(Region.US_WEST2)
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java:976]: define a local var for Region.US_WEST2

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java:33]: And is `VirtualMachineScaleSetSkuTypes.STANDARD_D2S_V3` same as this? If yes, use the predefined value.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java [hunk 3]
```diff
@@ -976,9 +973,9 @@ public void canEnableMSIOnVirtualMachineScaleSetWithoutRoleAssignment() throws E
...
                 .withExistingResourceGroup(resourceGroup)
-                .withSku(VirtualMachineScaleSetSkuTypes.STANDARD_A0)
+                .withSku(VirtualMachineScaleSetSkuTypes.fromSkuNameAndTier("Standard_D2s_v3", "Standard"))
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetOperationsTests.java:978]: STANDARD_D2S_V3 ?

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java [hunk 4]
```diff
@@ -398,17 +403,20 @@ public void canCreateVirtualMachineScaleSetWithLMSIAndEMSI() throws Exception {
                 "Expected role assignment not found for the virtual network for local identity"
                     + virtualMachineScaleSet.systemAssignedManagedServiceIdentityPrincipalId());
 
-        RoleAssignment assignment =
-            lookupRoleAssignmentUsingScopeAndRoleAsync(
+        RoleAssignment assignment;
+        if (!isPlaybackMode()) {
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineScaleSetEMSILMSIOperationsTests.java:407]: please add a comment, e.g. "principalId redacted"

File: sdk/resourcemanager/azure-resourcemanager-monitor/src/test/java/com/azure/resourcemanager/monitor/AlertsTests.java [hunk 5]
```diff
@@ -391,7 +388,9 @@ public void canCRUDMultipleResourceMetricAlerts() throws Exception {
             ma.refresh();
             Assertions.assertEquals(2, ma.scopes().size());
             Assertions.assertEquals(vm1.type(), ma.innerModel().targetResourceType());
-            Assertions.assertEquals(vm1.regionName(), ma.innerModel().targetResourceRegion());
+            if (!isPlaybackMode()) {
+                Assertions.assertEquals(vm1.regionName(), ma.innerModel().targetResourceRegion());
+            }
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-monitor/src/test/java/com/azure/resourcemanager/monitor/AlertsTests.java:393]: What happens in playback? I'd assume region won't be redacted?

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java [hunk 6]
```diff
@@ -337,12 +345,14 @@ public void print(TrafficManagerProfile profile) {
...
-                    .append("\n\t\t\tTarget resourceType: ")
-                    .append(endpoint.targetResourceType())
-                    .append("\n\t\t\tMonitor status: ")
+                    .append(endpoint.endpointType());
+                if (!isPlaybackMode) {
+                    info.append("\n\t\t\tTarget resourceId: ")
+                        .append(endpoint.targetAzureResourceId())
+                        .append("\n\t\t\tTarget resourceType: ")
+                        .append(endpoint.targetResourceType());
+                }
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java:354]: Why this needs the condition? I assume this is only for logging?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java]: Fixed in the new version.

File: sdk/resourcemanager/azure-resourcemanager-monitor/src/test/java/com/azure/resourcemanager/monitor/AlertsTests.java (hunk 5 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-monitor/src/test/java/com/azure/resourcemanager/monitor/AlertsTests.java]: The `targetResourceRegion` has been replace to `Sanitized` with latest TestProxy.

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java (hunk 6 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java]: The `targetResourceRegion` has been replace to `Sanitized` with latest TestProxy.

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java [hunk 7]
```diff
@@ -254,7 +255,9 @@ public void canCreateUpdateVirtualMachineWithEMSI() {
     }
 
     @Test
+    @LiveOnly
     public void canCreateVirtualMachineWithLMSIAndEMSI() {
+        // LiveOnly because "test timing out after latest test proxy update"
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java:260]: Is this time out for PLAYBACK only, or time out for both?

If PLAYBACK only, use `@DoNotRecord(skipInPlayback = true)` (`@LiveOnly` won't run in record mode)

File: sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java (hunk 6 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java]: Fixed what?

I mean, if this is just for logging, let it print. Why does it matter?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java]: The implementation of `targetResourceType` is as follows:
```
    @Override
    public TargetAzureResourceType targetResourceType() {
        return new TargetAzureResourceType(
            ResourceUtils.resourceProviderFromResourceId(targetAzureResourceId()),
            ResourceUtils.resourceTypeFromResourceId(targetAzureResourceId()));
    }
```
The specified ID `Sanitized` is not a valid Azure resource ID

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java]: I still didn't get it. Does call of `targetResourceType()` throws when id is not valid?

If it does not throw, invalid output like `null/null` does not matter much, if it is only some output.

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager/src/test/java/com/azure/resourcemanager/TestTrafficManager.java]: When the value of `targetResourceId` is replaced with `Sanitized`, an exception is thrown with the following message: `The specified ID Sanitized is not a valid Azure resource ID.`

File: sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java (hunk 7 above)
Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java]: Fixed in the new version.

Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java]: Answer? Is this time out for PLAYBACK only, or time out for both?

Comment (v-hongli1) [reply sdk/resourcemanager/azure-resourcemanager-compute/src/test/java/com/azure/resourcemanager/compute/VirtualMachineEMSILMSIOperationsTests.java]: This is time out for PLAYBACK only.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**